### ISSUE TITLE
GSConnect not showing app names after KDE Connect update on phone

### ISSUE THREAD

**Issue by phrogpilot73** [GSConnect/gnome-shell-extension-gsconnect#1211]
KDE Connect updated on my phone last night (version 1.18.0), running GS Connect version 45 (according to the about).

Notifications from my phone displayed on my laptop used to say something along the lines of "Kevo" or "Samsung Health". Now they say "com.android.kevo"

**Comment by andyholmes** [comment]
Please fill out the issue template if you would like help with this issue:

> ***ATTENTION**: GSConnect only supports the latest, stable version of GNOME. We are no longer accepting bug reports for previous versions.*

**Describe the bug**

A clear and concise description of what the bug is.

**Steps To Reproduce:**

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**

A clear and concise description of what you expected to happen.

**Screenshots**

If applicable, add screenshots to help explain your problem. You can drag-and-drop or cut-and-paste images directly into this edit window, to include them in your report.

**Support Log**

Please generate a support log ([Instructions](../wiki/Help#generate-support-log)) and paste any messages related to this issue between the two ``` lines below.

```

```

**System Details (please complete the following information):**

 - **GSConnect version:** [e.g. 35, 36, ... (displayed in the GSConnect "About" window)]
   - **Installed from:** [e.g. GNOME Extensions Website, GitHub, Package Manager, ...]
 - **GNOME/Shell version:** [e.g. 3.36, ...]
 - **Distro/Release:** [e.g. Ubuntu 18.04, Fedora 29, Arch, ...]

**GSConnect environment (if applicable):**

 - **Paired Device(s):** [e.g. Galaxy Note 8, Pixel 2, Honor 9, ...]
 - **KDE Connect app version:** [e.g. 1.12.7]
 - **Plugin(s):** [if the issue only occurs when using certain plugin(s)]

**Additional Notes:**

Add any additional information about the problem or your system.

**Comment by phrogpilot73** [comment]
**Describe the bug**

Notifications coming from my mobile device no longer show the app name in plain English. For example, what used to appear as "Ring" now appears as "com.ringapp". See attached screen shot.

**Steps To Reproduce:**

1. Connect phone to computer via GSConnect
2. Get notification from application
3. Notice application name

**Expected behavior**

Prior to KDE Connect updating on my phone, a plain English app name was included with the notification. I expect to see "Email" followed by the notification, or "WSJ" followed by the notification.

**Screenshots**

![Screenshot from 2021-11-28 10-12-02](https://user-images.githubusercontent.com/91998831/143773948-fc4002db-7a61-408f-a97f-11042033d664.png)

**Support Log**

GSConnect: 45 (user)
GJS:       16702
Session:   x11
OS:        Pop!_OS 21.04
--------------------------------------------------------------------------------
[0;1;39mHint: You are currently not seeing messages from other users and the system.[0m
[0;1;39m      Users in groups 'adm', 'systemd-journal' can see all messages.[0m
[0;1;39m      Pass -q to turn off this notice.[0m
-- Journal begins at Tue 2021-10-26 19:01:41 EDT, ends at Sun 2021-11-28 10:09:49 EST. --
Nov 28 10:09:13 dbus-daemon[7551]: [session uid=1000 pid=7551] Activating service name='org.gnome.gedit' requested by ':1.109' (uid=1000 pid=10386 comm="gjs /home/cory/.local/share/gnome-shell/extensions" label="unconfined")
Nov 28 10:09:13 dbus-daemon[7551]: [session uid=1000 pid=7551] Successfully activated service 'org.gnome.gedit'
Nov 28 10:09:14 gnome-shell[8003]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x3e000f8
Nov 28 10:09:19 onedriver[3843]: time="2021-11-28T10:09:19" level=debug msg="Fetching deltas from server." func="000058:fs.(*Cache).DeltaLoop()" file="delta.go:21"
Nov 28 10:09:19 onedriver[3843]: time="2021-11-28T10:09:19" level=info msg="Fetched 0 deltas." func="000058:fs.(*Cache).DeltaLoop()" file="delta.go:44"
Nov 28 10:09:19 onedriver[3843]: time="2021-11-28T10:09:19" level=debug msg="Serializing cache metadata to disk." func="000058:fs.(*Cache).SerializeAll()" file="cache.go:525"
Nov 28 10:09:42 org.gnome.Shell.Extensions.GSConnect[10150]: [/service/device.js:_readLoop:338]: SM-N986U: {
                                                               "id": 1638112181866,
                                                               "type": "kdeconnect.notification",
                                                               "body": {
                                                                 "actions": [
                                                                   "SNOOZE"
                                                                 ],
                                                                 "id": "0|com.ringapp|-1107923412|null|10422",
                                                                 "onlyOnce": false,
                                                                 "isClearable": true,
                                                                 "appName": "com.ringapp",
                                                                 "time": "1638112181606",
                                                                 "ticker": "There is a Person at your Front Door",
                                                                 "title": "There is a Person at your Front Door"
                                                               }
                                                             }
Nov 28 10:09:43 org.gnome.Shell.Extensions.GSConnect[10150]: [/service/device.js:_readLoop:338]: SM-N986U: {
                                                               "id": 1638112182278,
                                                               "type": "kdeconnect.notification",
                                                               "body": {
                                                                 "actions": [
                                                                   "SNOOZE"
                                                                 ],
                                                                 "id": "0|com.ringapp|-1107923412|null|10422",
                                                                 "onlyOnce": true,
                                                                 "isClearable": true,
                                                                 "appName": "com.ringapp",
                                                                 "time": "1638112182047",
                                                                 "ticker": "There is a Person at your Front Door",
                                                                 "title": "There is a Person at your Front Door"
                                                               }
                                                             }
Nov 28 10:09:49 onedriver[3843]: time="2021-11-28T10:09:49" level=debug msg="Fetching deltas from server." func="000058:fs.(*Cache).DeltaLoop()" file="delta.go:21"
Nov 28 10:09:49 onedriver[3843]: time="2021-11-28T10:09:49" level=info msg="Fetched 0 deltas." func="000058:fs.(*Cache).DeltaLoop()" file="delta.go:44"
Nov 28 10:09:49 onedriver[3843]: time="2021-11-28T10:09:49" level=debug msg="Serializing cache metadata to disk." func="000058:fs.(*Cache).SerializeAll()" file="cache.go:525"
**System Details (please complete the following information):**

    **GSConnect version:** 45
        **Installed from:** GNOME Extensions website
    **GNOME/Shell version:** 3.38.5
    **Distro/Release:** Pop!_OS 21.04

**GSConnect environment (if applicable):**

    **Paired Device(s):** Samsung Galaxy Note 20 Ultra
    **KDE Connect app version:** 1.18.0
    **Plugin(s):** N/A

**Additional Notes:**

None

**Comment by andyholmes** [comment]
Thanks for providing thorough logs! It looks like something may have changed or regressed in KDE Connect Android:

```js
{
  "id": 1638112181866,
  "type": "kdeconnect.notification",
  "body": {
    "actions": [
      "SNOOZE"
    ],
    "id": "0|com.ringapp|-1107923412|null|10422",
    "appName": "com.ringapp",            <-- The package ID is the fallback value for "appName"
    "ticker": "There is a Person at your Front Door",
    "title": "There is a Person at your Front Door",
    "time": "1638112181606",
    "onlyOnce": false,
    "isClearable": true
  }
}
```

When the `appName` field is unavailable KDE Connect Android [falls back to the package ID](https://github.com/KDE/kdeconnect-android/blob/7341cbd3d5218a7e593ea5f8ee7d1a39864a6cb7/src/org/kde/kdeconnect/Plugins/NotificationsPlugin/NotificationsPlugin.java#L231), although I don't see anything that has changed with that since v1.17.0. You'll have to open an upstream bug report for that, since I can't see why that would have regressed myself.

You can open a report [here](https://bugs.kde.org/describecomponents.cgi?product=kdeconnect) under `android-application`, and either reference this bug report or copy the packet from above to show what has happened. We can keep this issue open to track it on both sides.

The `undefined` in the notification is fair to call a bug in GSConnect, however, since it should check for that.

**Comment by rohmishra** [comment]
Reported to upstream at https://bugs.kde.org/show_bug.cgi?id=446266

**Comment by rohmishra** [comment]
https://invent.kde.org/network/kdeconnect-android/-/commit/beafdb26abe931ccc5671869dbb5141658915082 should fix this. Wait for 1.18.1 to be rolled out.

**Comment by andyholmes** [comment]
Looks like 1.18.1 has rolled out on Google Play, so closing now.

**Comment by phrogpilot73** [comment]
Any idea how soon after something hits Google Play that it hits FDroid?

**Comment by andyholmes** [comment]
I do not, AFAIK F-Droid has some tag-watching bot that pulls and builds occasionally.
